### ISSUE TITLE
Cop idea: DSL Method Calls

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12354]
## Description

Always omit parentheses for methods that are part of an internal DSL (e.g., Rake, Rails, RSpec):

https://rubystyle.guide/#no-dsl-decorating

## Behavior

```ruby
class Person < ActiveRecord::Base
  # bad
  attr_reader(:name, :age)
  # good
  attr_reader :name, :age

  # bad
  validates(:name, { presence: true, length: { within: 1..10 } })
  # good
  validates :name, presence: true, length: { within: 1..10 }
end
```

**Comment by Earlopain** [comment]
There is a case to be considered when using splats.

```rb
attr_reader *BAR
```

This requires braces since it would otherwise be a ruby warning and also be caught by `Lint/AmbiguousOperator`. See #12212
